tests/libkubevirt/config/kvconfig.go

```diff
@@ -120,10 +120,19 @@ func ExpectResourceVersionToBeLessEqualThanConfigVersion(resourceVersion, config
 	return true
 }
 
-func waitForConfigToBePropagated(resourceVersion string) {
-	WaitForConfigToBePropagatedToComponent("kubevirt.io=virt-controller", resourceVersion, ExpectResourceVersionToBeLessEqualThanConfigVersion, 10*time.Second)
-	WaitForConfigToBePropagatedToComponent("kubevirt.io=virt-api", resourceVersion, ExpectResourceVersionToBeLessEqualThanConfigVersion, 10*time.Second)
-	WaitForConfigToBePropagatedToComponent("kubevirt.io=virt-handler", resourceVersion, ExpectResourceVersionToBeLessEqualThanConfigVersion, 10*time.Second)
+func waitForConfigToBePropagated(desiredGeneration int64) {
+	virtClient := kubevirt.Client()
+
+	Eventually(func() bool {
+		kv := libkubevirt.GetCurrentKv(virtClient)
+		observedGeneration := *kv.Status.ObservedGeneration
+
+		if observedGeneration >= desiredGeneration {
+			return true
+		}
+
+		return false
+	}, 120*time.Second, 2*time.Second).Should(BeTrue())
```

Comment:
```suggestion
func waitForConfigToBePropagated(desiredGeneration int64) {
	virtClient := kubevirt.Client()
	Eventually(func(g Gomega) {
		kv := libkubevirt.GetCurrentKv(virtClient)
		Expect(kv.Status.ObservedGeneration).ToNot(BeNil())
		Expect(kv.Status.ObservedGeneration).To(BeNumerically(">=", desiredGeneration))
	}, 120*time.Second, 2*time.Second).Should(Succeed())
```

Reply:
Also, are you sure this needs to be a separate func? IMO it could be inlined into `UpdateKubeVirtConfigValueAndWait` because it is simple enough.

Reply:
Thanks, thats a good point :+1: